app/src/main/java/com/ichi2/apisample/ui/MainActivity.java

```diff
@@ -795,7 +796,11 @@ public void run() {
                 intervalKeys = newMi.intervals;
                 afterAdding = true;
                 refreshFilenames();
-                savedInstruments.add(newMi.instrument);
+                String addedInstrument = newMi.instrument;
+                savedInstruments.add(addedInstrument);
+                if (!savedInstruments.contains(addedInstrument)) {
```

Comment:
Looks like a typo or some invalid logic, is not it?
probably should be "savedInstrumentsAdapter.contains(addedInstrument)" ?

Reply:
Nice catch. Fixed